lite/valor_lite/detection/manager.py

```diff
@@ -817,45 +908,129 @@ def add_data(
             # cache labels and annotations
             keyed_groundtruths = defaultdict(list)
             keyed_predictions = defaultdict(list)
+
             for gidx, gann in enumerate(detection.groundtruths):
-                self._evaluator.groundtruth_examples[uid_index][
-                    gidx
-                ] = np.array(gann.extrema)
+                if not isinstance(gann, annotation_type):
+                    raise ValueError(
+                        f"Expected {annotation_type}, but annotation is of type {type(gann)}."
+                    )
+
+                if annotation_type == BoundingBox:
+                    self._evaluator.groundtruth_examples[uid_index][
+                        gidx
+                    ] = _get_annotation_representation(obj=gann)
+
                 for glabel in gann.labels:
                     label_idx, label_key_idx = self._add_label(glabel)
                     self.groundtruth_count[label_idx][uid_index] += 1
+                    representation = _get_annotation_representation(obj=gann)
```

Comment:
This should be moved out of this loop as it is just performing the same op for all labels.

Reply:
The examples call can share the results of a single call too.